app_radio_module/src/transmitter.c

```diff
@@ -81,6 +90,36 @@ static void lora_tx_task(void *, void *, void *) {
     }
 }
 
+typedef struct {
+    double latitude;
+    double longitude;
+    double altitude;
+} gnss_data_simple;
+
+static void gnss_data_cb(const struct device *dev, const struct gnss_data *data) {
+    if (!ready_to_tx) {
+        return; // timer hasnt expired yet
+    }
+    l_lora_packet_t packet = {0};
+    int port_num = RADIO_MODULE_GNSS_DATA_PORT + RADIO_MODULE_BASE_PORT;
+    packet.port = port_num;
+    packet.payload_len = sizeof(gnss_data_simple);
+
+    gnss_data_simple gnss_data = {0};
+    gnss_data.latitude = data->nav_data.latitude / L_GNSS_LATITUDE_DIVISION_FACTOR;
+    gnss_data.longitude = data->nav_data.longitude / L_GNSS_LONGITUDE_DIVISION_FACTOR;
+    gnss_data.altitude = data->nav_data.altitude / L_GNSS_ALTITUDE_DIVISION_FACTOR;
```

Comment:
Explicit cast these too to

Reply:
resolved https://github.com/RIT-Launch-Initiative/FSW/pull/85/commits/7c343f3c788021bde2044040077e994b33921dd1